### PR TITLE
Add prometheus metrics to CSI external-snapshotter using new csi-lib-utils library

### PR DESCRIPTION
**Pull request by saad-ali**
**What type of PR is this?**

/kind feature

**What this PR does / why we need it**:

This PR adds metrics to the external-snapshotter CSI gRPC calls:

* `csi_sidecar_operations_seconds` [ALPHA] Container Storage Interface operation duration with gRPC error code status total

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
Add prometheus metrics to CSI external-snapshotter under the /metrics endpoint. This can be enabled via the "--metrics-address" and "--metrics-path" options.
```

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -4,19 +4,59 @@ go 1.12
...
+
+replace k8s.io/cluster-bootstrap => k8s.io/cluster-bootstrap v0.17.0
+
+replace k8s.io/code-generator => k8s.io/code-generator v0.17.1-beta.0
+
+replace k8s.io/component-base => k8s.io/component-base v0.17.0
+
+replace k8s.io/cri-api => k8s.io/cri-api v0.17.1-beta.0
+
+replace k8s.io/csi-translation-lib => k8s.io/csi-translation-lib v0.17.0
```
Comment (msau42) [inline go.mod]: what actually uses csi-translation-lib here?

Comment (saad-ali) [reply go.mod]: Yep, ran `go mod tidy`:

```SHELL
$ go mod tidy
$ git status
On branch addMetricsUsingLib
nothing to commit, working tree clean
```

Doesn't look like anything is using it. My guess is replace does not get removed by tidy. Let me try to remove it manually.

Comment (saad-ali) [reply go.mod]: https://github.com/kubernetes-csi/external-snapshotter/pull/228

File: README.md [hunk 2]
```diff
@@ -85,6 +85,10 @@ Install CSI Driver:
 
 * `--leader-election-namespace <namespace>`: The namespace where the leader election resource exists. Defaults to the pod namespace if not set.
 
+* `--metrics-address`: The TCP network address address where the prometheus metrics endpoint will run (example: `:8080` which corresponds to port 8080 on local host). The default is empty string, which means metrics endpoint is disabled.
```
Comment (xing-yang) [inline README.md]: nit: duplicate "address".

File: cmd/csi-snapshotter/main.go [hunk 3]
```diff
@@ -66,6 +67,9 @@ var (
 
 	leaderElection          = flag.Bool("leader-election", false, "Enables leader election.")
 	leaderElectionNamespace = flag.String("leader-election-namespace", "", "The namespace where the leader election resource exists. Defaults to the pod namespace if not set.")
+
+	metricsAddress = flag.String("metrics-address", "", "The TCP network address address where the prometheus metrics endpoint will listen (example: `:8080`). The default is empty string, which means metrics endpoint is disabled.")
```
Comment (xing-yang) [inline cmd/csi-snapshotter/main.go]: duplicate "address".

Comment (saad-ali) [reply cmd/csi-snapshotter/main.go]: Good catch. https://github.com/kubernetes-csi/external-snapshotter/pull/229

**Comment by saad-ali** [comment]
> So this PR has fixed this issue, right? #142

Yes.

Fixes #142

**Comment by verult** [comment]
@mattcary @saikat-royc

**Pull request merged**